consensus.go

```diff
@@ -113,6 +113,9 @@ type Backend interface {
 	// Hash hashes the proposal bytes
 	Hash(p []byte) []byte
 
+	// Init is used to signal the backend that a new round is going to start.
+	Init()
```

Comment:
Have you considered maybe that this function returns an `error` object or `bool` in case it fails? Not sure how trivial logic within it would be and is there any scenario where `Init` function fails and it is needed to be handled somehow from the "outside"?